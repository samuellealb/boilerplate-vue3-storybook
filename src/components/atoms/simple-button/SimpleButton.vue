<template>
  <div :class="classes">
    {{ title }}
  </div>
</template>

<script>
import { reactive, computed } from 'vue';
import './simple-button.css';

export default {
  props: {
    title: {
      type: String,
      required: true,
    },
    variant: {
      type: String,
      default: 'default',
      validator: function (value) {
        return ['default', 'light', 'dark'].indexOf(value) !== -1;
      },
    },
    size: {
      type: String,
      default: 'medium',
      validator: function (value) {
        return ['small', 'medium', 'large'].indexOf(value) !== -1;
      },
    },
  },
  setup(props) {
    props = reactive(props)
    const classes = computed(() => ({
      'simple-button': true,
      [`simple-button--${props.variant || 'default'}`]: true, 
      [`simple-button--${props.size || 'medium'}`]: true,      
    }));
    return { classes }
  },
}
</script>
