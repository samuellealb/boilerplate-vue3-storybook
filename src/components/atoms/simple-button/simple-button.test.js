import { mount } from '@vue/test-utils'
import { test, expect } from 'vitest'
import SimpleButton from './SimpleButton.vue'

test('mount component', async () => {
  expect(SimpleButton).toBeTruthy()

  const defaultButton = mount(SimpleButton, {
    props: {
      title: 'Default Button',
      variant: 'default',
      size: 'medium'
    }
  })
  expect(defaultButton.html()).toMatchSnapshot()

  const lightButton = mount(SimpleButton, {
    props: {
      title: 'Light Button',
      variant: 'light',
      size: 'medium'
    }
  })
  expect(lightButton.html()).toMatchSnapshot()

  const darkButton = mount(SimpleButton, {
    props: {
      title: 'Dark Button',
      variant: 'dark',
      size: 'medium'
    }
  })
  expect(darkButton.html()).toMatchSnapshot()

  const smallButton = mount(SimpleButton, {
    props: {
      title: 'Small Button',
      variant: 'default',
      size: 'small'
    }
  })
  expect(smallButton.html()).toMatchSnapshot()

  const largeButton = mount(SimpleButton, {
    props: {
      title: 'Large Button',
      variant: 'default',
      size: 'large'
    }
  })
  expect(largeButton.html()).toMatchSnapshot()

})