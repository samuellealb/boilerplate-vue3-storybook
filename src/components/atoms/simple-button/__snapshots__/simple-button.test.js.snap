// Vitest Snapshot v1

exports[`mount component 1`] = `"<button class=\\"simple-button simple-button--default simple-button--medium\\">Default Button</button>"`;

exports[`mount component 2`] = `"<button class=\\"simple-button simple-button--light simple-button--medium\\">Light Button</button>"`;

exports[`mount component 3`] = `"<button class=\\"simple-button simple-button--dark simple-button--medium\\">Dark Button</button>"`;

exports[`mount component 4`] = `"<button class=\\"simple-button simple-button--default simple-button--small\\">Small Button</button>"`;

exports[`mount component 5`] = `"<button class=\\"simple-button simple-button--default simple-button--large\\">Large Button</button>"`;
